javascript/ql/src/Security/CWE-022/ZipSlipGood.js

```diff
@@ -0,0 +1,14 @@
+const fs = require('fs');
+const unzip = require('unzip');
+
+fs.createReadStream('archive.zip')
+  .pipe(unzip.Parse())
+  .on('entry', entry => {
+    const fileName = entry.path;
+    if (fileName.indexOf('..') == -1) {
+      entry.pipe(fs.createWriteStream(fileName));
+    }
+    else {
+      console.log('skipping bad path', fileName);
```

Comment:
Likewise, there's normally something to indicate that this is good practice.

Reply:
Done.